tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/TaskImpl.java

```diff
@@ -1027,7 +1027,7 @@ public void transition(TaskImpl task, TaskEvent event) {
       TaskAttempt earliestUnfinishedAttempt = null;
       for (TaskAttempt ta : task.attempts.values()) {
         // find the oldest running attempt
-        if (!ta.isFinished()) {
+        if (!ta.isFinished() && task.commitAttempt == null) {
```

Comment:
Thanks for this patch @Chyler. This has the correct behavior, but the LOG.info below may give the wrong reason. Can this be improved to log the two cases separately?

Reply:
+1

Reply:
done